### PR TITLE
jwt implementation, log4j2 and code refactoring

### PR DESCRIPTION
**Pull request by nithin554**
Big Changes!!! Please review carefully....

- Request and Response structure changes ( for login and signup endpoints )
- JWT Implementation
- Added log4j2 logger
- Code Refactoring
- Minor Bug Fixes

### CODE REVIEW

**Comment by nithin554** [comment]
@pallasaisrujan @saikrishna-dev review fast

File: src/main/java/com/backend/webapp/controller/LoginController.java [hunk 1]
```diff
@@ -28,23 +35,26 @@ public class LoginController {
...
-            if (null != user && null != user.getUserId()
-                    && user.getUserId().equalsIgnoreCase(loginRequest.getUsername())
+            Users user = usersRepository.findByEmail(loginRequest.getEmail());
+            if (null != user && null != user.getEmail() && user.getEmail().equalsIgnoreCase(loginRequest.getEmail())
                     && null != user.getPasswordHash()) {
-                String decryptedPassword = EncryptionUtil.decryptData(secretManagerTemplate, user.getPasswordHash());
-                if (password.equalsIgnoreCase(decryptedPassword)) {
-                    return ResponseEntity.ok("Success");
+                if (password
+                        .equalsIgnoreCase(EncryptionUtil.decryptData(secretManagerTemplate, user.getPasswordHash()))) {
```
Comment (saikrishna-dev) [inline src/main/java/com/backend/webapp/controller/LoginController.java:48]: remove `equalsIgnoreCase `while validating the password after decrypting ?

File: src/main/java/com/backend/webapp/mapper/RequestMapper.java [hunk 2]
```diff
@@ -14,7 +14,7 @@ public static Users mapToUsers(SignupRequest signupRequest) {
         user.setEmail(signupRequest.getEmail());
         user.setPasswordHash(signupRequest.getPasswordHash());
         user.setRole(signupRequest.getRole());
-        user.setUserId(signupRequest.getUserId());
+        user.setFullName(signupRequest.getFullName());
```
Comment (saikrishna-dev) [inline src/main/java/com/backend/webapp/mapper/RequestMapper.java:17]: `getFullName()` Where is this defined??

Comment (saikrishna-dev) [reply src/main/java/com/backend/webapp/mapper/RequestMapper.java:17]: same with `setFullName`

**Review by saikrishna-dev — CHANGES_REQUESTED** [review]

File: src/main/java/com/backend/webapp/controller/LoginController.java (hunk 1 above)
Comment (nithin554) [reply src/main/java/com/backend/webapp/controller/LoginController.java:48]: I will remove this

File: src/main/java/com/backend/webapp/mapper/RequestMapper.java (hunk 2 above)
Comment (nithin554) [reply src/main/java/com/backend/webapp/mapper/RequestMapper.java:17]: check parent class

File: src/main/java/com/backend/webapp/controller/LoginController.java (hunk 1 above)
Comment (nithin554) [reply src/main/java/com/backend/webapp/controller/LoginController.java]: resolved

**Review by saikrishna-dev — APPROVED** [review]

**Pull request merged**